### ISSUE TITLE
Some app windows resize oddly when monitor focus changes

### ISSUE THREAD

**Issue by crispinb** [paperwm/PaperWM#498]
**Describe the bug**

In at least a couple of terminal apps (wezterm & kitty), switching focus to a different monitor sometimes causes the app's windows (on the original monitor) to resize, causing  their text to distort. 

**To Reproduce**
(in a multi-monitor configuration)
1. open a window in an app affected by this issue (eg. wezterm)
2. change focus to another monitor (using either keyboard or mouse)

**Expected behavior**

Nothing should happen other than monitor & window focus change,
but the new window on the monitor losing focus changes to a new size, distorting
text and often leaving an outline of the original size.

The window goes back to its original size when focus returns to the monitor it is on.

**Screenshots**
First ss shows the wezterm window with the monitor it is on having focus
Second ss shows how it looks when focus is moved to another monitor

![original _res](https://user-images.githubusercontent.com/630851/226761858-91a1a4e2-8a38-49a9-9c5b-697cf59afaaa.png)
![new_size](https://user-images.githubusercontent.com/630851/226763120-39fb9d8d-122d-4b25-abc6-1513fd335877.png)

**System information:**

```
Distribution: Fedora Linux
GNOME Shell 43.3
PaperWM branch/tag: develop
PaperWM commit: 7fa25a6072a8ccd51ecbf154d030f23d2ed98d4e
Enabled extensions:
- impatience@gfxmonk.net
- clipboard-history@alexsaveau.dev
- paperwm@hedning:matrix.org
- vertical-overview@RensAlthuis.github.com
- background-logo@fedorahosted.org
- appindicatorsupport@rgcjonas.gmail.com
```

**Additional context**

I think I've only experienced this with terminal apps - wezterm and kitty, but the latter only infrequently. I would intuitively have attributed this problem to those apps and submitted it to their repos, except for the fact that it only happens when PaperWM is enabled. 

This doesn't always happen, but I haven't found an obvious discriminating factor.

The windows never resize (nor does text distort) when the terminal window is full-screen on the original monitor.

**Comment by jtaala** [comment]
Thanks @crispinb, there's been some work re multi-monitors recently (with #482).  Looks like you're currently not on the latest commit.  Can you update to latest (e.g. with a `git pull`), logout/login and confirm the behaviour is still happening?

**Comment by crispinb** [comment]
Yep, I've pulled the latest HEAD and can confirm it still happens.  It's not a huge deal as only a very small number of apps seem to be affected: `wezterm` the most, and `kitty` infrequently (actually I haven't been able to get it to happen with kitty recently, so perhaps something has changed there). I'm more attached to PaperWM than to any particular terminal app.

This screencast illustrates perhaps more clearly what's going on. All that's triggering that terminal window resize is flipping between two monitors with the relevant PaperWM keyboard shortcut.


[Screencast from 2023-03-22 18-01-46.webm](https://user-images.githubusercontent.com/630851/226843104-795bee94-1573-4cb1-9066-38080ede511a.webm)

**Comment by jtaala** [comment]
Thanks @crispinb for checking!

**Comment by crispinb** [comment]
There's some kind of resizing history/path dependence going on, for example:
- I can reliably trigger the issue in wezterm by full-screening (not maximising) then restoring its window, then switching monitors
- if a wezterm window is in the problematic state, I can usually fix it by resizing its window a few times. This seems a bit random - but usually a few `Super R`'s does the job.

**Comment by jtaala** [comment]
Hmm - I haven't been able to reproduce this.  I've installed wezterm and have two instances on one of my monitors.

Wait, @crispinb, are you on wayland or X11?  Or does this happen on both?

Could rerun you `gather-system-info.sh` and paste output here again? (just want to confirm I'm on the exact commit as you too).

**Comment by jtaala** [comment]
Lastly, are both your monitors the same resolution? (or do they have different resolutions?).

**Comment by Lythenas** [comment]
I managed to (kind of) reproduce it. Although for me the issue doesn't happen when focusing monitors it simply happens when starting a second instance and also the first time unfocusing the new instance.

[Screencast from 2023-03-25 12-53-44.webm](https://user-images.githubusercontent.com/32217236/227715966-0d8dc99a-43bd-4346-8855-604d14159af3.webm)

Something similar also sometimes happens with firefox for me. Although it only occurs when it has been running for a long time and it is fixed after restarting firefox. So I haven't bothered looking into it.

Also I noticed that (at least for me) the issue only happens while switching (i.e. while the overview window is open (which can be disabled), until you let go of super). It also happens for me if I only have a single floating window of wezterm.

My understanding is that during switching we only display actor clones instead of the real windows. I think this is to make animations work.

I'm not familiar with that part of the code but maybe something is going on there. But it is also strange that it only happens for a small number of apps.

**Comment by Lythenas** [comment]
Also I noticed that (at least for me) the issue only happens while switching (i.e. while the overview window is open (which can be disabled), until you let go of super). It also happens for me if I only have a single floating window of wezterm.

My understanding is that during switching we only display actor clones instead of the real windows. I think this is to make animations work.

I'm not familiar with that part of the code but maybe something is going on there. But it is also strange that it only happens for a small number of apps.

**Comment by Lythenas** [comment]
Also after using it a bit I noticed that when wezterm is unfocused it seems to resize back to it's previous size. E.g. the size before hitting super+f. If you resize it manually with the mouse then there is no issue (maybe you also can't notice it because the previous size would be just 1px smaller).

Also wezterm does not respect the keyboard layout that is selected in gnome... :sweat_smile: 

Maybe this is also related to wez/wezterm/issues/695

**Comment by crispinb** [comment]
> Wait, @crispinb, are you on wayland or X11? Or does this happen on both?
> 
> Could rerun you `gather-system-info.sh` and paste output here again? (just want to confirm I'm on the exact commit as you too).

I'm on Wayland. Here's the system info:

Distribution: Fedora Linux
GNOME Shell 43.3
PaperWM branch/tag: develop
PaperWM commit: 435fb826de8c79dee20de73425642111d0622b20
Enabled extensions:
- impatience@gfxmonk.net
- clipboard-history@alexsaveau.dev
- paperwm@hedning:matrix.org
- background-logo@fedorahosted.org
- appindicatorsupport@rgcjonas.gmail.com


> Lastly, are both your monitors the same resolution? (or do they have different resolutions?).

Different - this is a laptop with internal monitor at 2560 x 1440 and external 3840 x 2160.

I've been using kitty exclusively for the past few days, and although I have experienced the same with it infrequently in the past, it hasn't happened again this time round. So perhaps, this is either a wezterm-exclusive issue or very nearly so. Perhaps it's not worth too much of your time given that. There's no shortage of excellent Linux terminal apps.

**Comment by crispinb** [comment]
> Also I noticed that (at least for me) the issue only happens while switching (i.e. while the overview window is open (which can be disabled), until you let go of super).

This is also the case for me when the issue occurs with wezterm windows on the same monitor. But when they're on different monitors, the badly-sized one returns to its correct size when the monitor it's on is focused (keyboard or mouse), and mis-sizes again when focus goes back to the other monitor.

**Comment by crispinb** [comment]
> If you resize it manually with the mouse then there is no issue

Ah yes I hadn't noticed that. Perhaps related to how I am usually able to stop this happening by hitting `Super R` a few times on the errant window

**Comment by stefangweichinger** [comment]
I also hit this with Tilix, on Fedora 39. 2 different monitors. Will check the Super-R "fix".
